### PR TITLE
MOSIP-24773 Fixed Download card issue.

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utility.java [hunk 1]
```diff
@@ -275,16 +276,23 @@ public Map<String, Object> getMailingAttributes(String id, Set<String> templateL
...
+	private Set<String> getPreferredLanguage(JSONObject demographicIdentity) {
 		String preferredLang = null;
 		String preferredLangAttribute = env.getProperty("mosip.default.user-preferred-language-attribute");
 		if (!StringUtils.isBlank(preferredLangAttribute)) {
 			Object object = demographicIdentity.get(preferredLangAttribute);
 			if(object!=null) {
 				preferredLang = String.valueOf(object);
+				if(preferredLang.contains(ResidentConstants.COMMA)){
+					String[] preferredLangArray = preferredLang.split(ResidentConstants.COMMA);
+					return new HashSet<>(Arrays.asList(preferredLangArray));
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/Utility.java]: I think you can use Arrays.asSet method. Please check.

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utility.java [hunk 2]
```diff
@@ -275,16 +276,23 @@ public Map<String, Object> getMailingAttributes(String id, Set<String> templateL
...
+				}
 			}
 		}
-		return preferredLang;
+		if(preferredLang!=null){
+			return new HashSet<>(List.of(preferredLang));
+		}
+		return new HashSet<>(List.of());
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/Utility.java]: You can use Set.of method.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utility.java (hunk 1 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/Utility.java]: There are no such method Arrays.asSet
simplified above
`return Arrays.stream(preferredLangArray).collect(Collectors.toSet());`

File: resident/resident-service/src/main/java/io/mosip/resident/util/Utility.java (hunk 2 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/Utility.java]: Done.

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**